src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs

```diff
@@ -613,5 +667,34 @@ internal ImmutableArray<CompletionProvider> GetAllProviders(ImmutableHashSet<str
                     cancellationToken);
             }
         }
+
+        private class ProjectCompletionProvider
```

Comment:
Nit: move to a separate file. This would also make it consistent with existing AbstractProjectExtensionProvider implementations.